src/sickle.ts

```diff
@@ -58,10 +62,49 @@ class Annotator {
         this.maybeVisitType((<ts.VariableDeclaration>node).type);
         this.writeNode(node);
         break;
+      case ts.SyntaxKind.ClassDeclaration: {
+        let classNode = <ts.ClassDeclaration>node;
+        let hasCtor = classNode.members.some((e) => e.kind === ts.SyntaxKind.Constructor);
+        if (hasCtor) {
+          this.writeNode(classNode);
+          break;
+        }
+        // Emit a synthetic ctor.
+        // TODO(martinprobst): Handle inherited parent ctors.
+        this.writeTextBetween(classNode, classNode.getFullStart(), classNode.getLastToken());
+        this.emit('constructor() {\n');
+        this.emitStubDeclarations(classNode, []);
+        this.emit('}\n');
+        this.writeNode(classNode.getLastToken());
+        break;
+      }
+      case ts.SyntaxKind.Constructor: {
+        let ctor = <ts.ConstructorDeclaration>node;
+        this.writeTextBetween(ctor, ctor.getFullStart(), ctor.body);
+        if (ctor.body.statements.length) {
+          this.writeTextBetween(ctor.body, ctor.body.getFullStart(), ctor.body.statements[0]);
+        } else {
+          this.emit(' {');
```

Comment:
is that even possible, I am reading this as a constructor with not body, i.e. ```class A{ constructor(); }```. Typescript playground says thats invalid. Otherwise, we need a test of such a beast.